metadata/updater/updater.go

```diff
@@ -620,12 +588,8 @@ func (update *Updater) persistMetadata(roleName string, data []byte) error {
 	}
 	// caching enabled, proceed with persisting the metadata locally
 	fileName := filepath.Join(update.cfg.LocalMetadataDir, fmt.Sprintf("%s.json", url.QueryEscape(roleName)))
-	cwd, err := os.Getwd()
-	if err != nil {
-		return err
-	}
 	// create a temporary file
-	file, err := os.CreateTemp(cwd, "tuf_tmp")
+	file, err := os.CreateTemp(update.cfg.LocalMetadataDir, "tuf_tmp")
```

Comment:
Just to be on the safe side, could we please use a new, random, temp filename every time?

Reply:
That is how `os.CreateTemp` works: https://pkg.go.dev/os#CreateTemp

Reply:
yep, as an example [a previous test run](https://github.com/theupdateframework/go-tuf/actions/runs/9470099707/job/26090235976#step:5:14) generated `tuf_tmp1013966631` as the temp file name

Reply:
My bad, should have consulted the docs. (Knew this functionality almost certainly must exist in Go, but not exactly where.) Thanks!